great_expectations/expectations/expectation.py

```diff
@@ -368,7 +368,7 @@ def __eq__(self, other: object) -> bool:
 
         # rendered_content is derived from the rest of the expectation, and can/should
         # be excluded from equality checks
-        exclude: set[str] = {"rendered_content"}
+        exclude: set[str] = {"rendered_content", "id"}
```

Comment:
Legacy equality checks (i.e. using `configuration`) exclude `id`

Reply:
I don't love it but if we look at the script I mentioned in Slack, we get different behavior based on this:

```python
suite = context.suites.add(gx.ExpectationSuite("chetan-test-2024-11-13-v21"))

for i in range(10):
    e = gxe.ExpectColumnValuesToBeBetween(column="age", min_value=0, max_value=100)
    suite.add_expectation(e)

print(len(suite.expectations)) # How many should we see?
```

We historically have gotten 1 from this but making `id` part of the `__eq__` check gets us 10.